src/components/Badge.tsx

```diff
@@ -0,0 +1,67 @@
+import React from 'react';
+import {GestureResponderEvent, PressableStateCallbackType, StyleProp, TextStyle, View, ViewStyle} from 'react-native';
+import styles from '@styles/styles';
+import * as StyleUtils from '@styles/StyleUtils';
+import CONST from '@src/CONST';
+import PressableWithoutFeedback from './Pressable/PressableWithoutFeedback';
+import Text from './Text';
+
+type BadgeProps = {
+    /** Is Success type */
+    success?: boolean;
+
+    /** Is Error type */
+    error?: boolean;
+
+    /** Whether badge is clickable */
+    pressable?: boolean;
+
+    /** Text to display in the Badge */
+    text: string;
+
+    /** Text to display in the Badge */
+    environment?: string;
+
+    /** Styles for Badge */
+    badgeStyles?: StyleProp<ViewStyle>;
+
+    /** Styles for Badge Text */
+    textStyles?: StyleProp<TextStyle>;
+
+    /** Callback to be called on onPress */
+    onPress: (event?: GestureResponderEvent | KeyboardEvent) => void;
+};
+
+function Badge({success = false, error = false, pressable = false, text, environment = CONST.ENVIRONMENT.DEV, badgeStyles, textStyles, onPress = () => {}}: BadgeProps) {
+    const textColorStyles = success || error ? styles.textWhite : undefined;
```

Comment:
`textColorStyles` shouldn't be a style? With this logic is boolean or object style. Maybe worth creating a function and return undefined when the success prop is true? Nested inline conditions can be confused

Reply:
It will be an style object if `success` or `error` is `true`, otherwise it will be `undefined`, TS is inferring correctly. 🙂 

<img width="553" alt="Screenshot 2023-11-13 at 09 48 13" src="https://github.com/Expensify/App/assets/20051562/772350b4-ef07-420d-a1c4-63ac7bf78da1">